fstest/fstests/fstests.go

```diff
@@ -1205,6 +1206,46 @@ func Run(t *testing.T, opt *Opt) {
 				}, fs.GetModifyWindow(ctx, f))
 			})
 
+			// TestFsListRootedSubdir tests putting and listing with an Fs that is rooted at a subdirectory 2 levels down
+			TestFsListRootedSubdir := func(t *testing.T) {
+				skipIfNotOk(t)
+				newF, err := cache.Get(ctx, subRemoteName+"/hello? sausage/êé")
+				assert.NoError(t, err)
+				nestedFile := fstest.Item{
+					ModTime: fstest.Time("2001-02-03T04:05:06.499999999Z"),
+					Path:    "a/b/c/d/e.txt",
+				}
+				_, _ = testPut(ctx, t, newF, &nestedFile)
+				depth := 10
+
+				objs, dirs, err := walk.GetAll(ctx, newF, "", true, depth)
```

Comment:
Can you use `fstest.CheckListing` to see whether everything is OK? That does retries and exponential backoff which `walk.GetAll` doesn't which is often needed on eventually consistent backends.

Reply:
Ok!